docs/source/examples/Image_reconstruction.rst

```diff
@@ -0,0 +1,301 @@
+    plt.show()
+
+.. image:: /_static/img/image_reconstruction/mnist_batch.png
+
+Reconstructed images after 3 epochs of training
+===============================================
+
+In order to visually evaluate the quality of the results, here, we show the reconstruction results of the samples shown above.
+
+.. code-block:: python
+
+    # Calculating predictions of the trained network on a batch
+    outputs = torch.tensor(model.predict_on_batch(inputs))
+    output_grid = make_grid(outputs)
+    fig = plt.figure(figsize=(10, 10))
+    out = output_grid.numpy().transpose((1, 2, 0))
+    plt.imshow(out)
+    plt.show()
+
+.. image:: /_static/img/image_reconstruction/mnist_3epoch.png
+
+Evaluation
+==========
+
+One of the strong and useful tools in Poutyne is the evaluate methods, which not only provide you with the evaluation metrics but also provide the ground truths and the predictions if the related arguments have been set to True (as below).
+
+.. code-block:: python
+
+    # evaluating thetrained network on test data
+    loss, predictions, ground_truth = model.evaluate_generator(test_dataloader, return_pred=True, return_ground_truth=True)
+
+In most computer vision applications, such as image reconstruction, it is very important to check the network's failures (or abilities, vice versa). The following part shows an input and a reconstructed image, as well as its reconstruction error map. The reconstruction error map shows which part of the image has not been reconstructed accurately.
+
+.. code-block:: python
+
+    sample_number = 2   # a sample from test dataset
+    
+    sample = ground_truth[sample_number][0]
+    sample_prediction_result_3epochs = predictions[sample_number][0]
+    
+    recunstruction_error_map_3epochs = sample - sample_prediction_result_3epochs  #reconstruction error map
+    fig, (ax1, ax2, ax3) = plt.subplots(1,3)
+    ax1.imshow(sample)
+    ax1.set_title('sample')
+    
+    ax2.imshow(sample_prediction_result_3epochs)
+    ax2.set_title('prediction')
+    
+    ax3.imshow(np.abs(recunstruction_error_map_3epochs))
+    ax3.set_title('reconstruction error')
+    plt.show()
+
+.. image:: /_static/img/image_reconstruction/rec_error_3epoch.png
+
+Resuming the training for more epochs
+=====================================
+
+If we find the past epochs not enough, Poutyne allows you to resume the training from the last done epoch, as below. Please note that, in the `callbacks` that we defined before, since we did not set the `restore_best argument` to `True`, our model stays at the last epoch after finishing the first part of the training. Hence, by setting the `initial_epoch` to the last epoch of the previous training, we can resume our training for more epoch, using the last saved weights. Otherwise, to benefit from the saved best weights, all you need is to set the `restore_best` argument to True and resume your training from the epoch number corresponding to the best weights saved by the callbacks.
```

Comment:
```suggestion
If we find the past epochs not enough, Poutyne allows you to resume the training from the last done epoch, as below. Please note that, in the `callbacks` that we defined before, since we did not set the `restore_best` argument in `ModelCheckpoint` to `True`, our model stays at the last epoch after finishing the first part of the training. Hence, by setting the `initial_epoch` to the last epoch of the previous training, we can resume our training for more epochs, using the last state of the neural network.
```